.github/workflows/dns-preview.yml

```diff
@@ -0,0 +1,27 @@
+---
+name: Test/Preview DNS changes
+
+on:
+  pull_request:
+
+permissions: {}
+
+jobs:
+  dnscontrol:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4.2.2
+      - uses: cachix/install-nix-action@v31
+      - name: dnscontrol check
+        working-directory: ./dns/
+        run: |
+          nix run --inputs-from . nixpkgs#dnscontrol -- check
+      - name: dnscontrol push
+        if: github.repository == 'nixos/infra'
+        env:
+          NETLIFY_TOKEN: "${{ secrets.NETLIFY_TOKEN }}"
```

Comment:
This looks like the same token as the deploy. Any chance we could use a read only token here instead?

Reply:
No scoped token, which is why I consider netlify a bad provider.

Reply:
For avoidance of doubt: at least GitHub Actions doesn't provide secrets to forked repos, which means the actual risk of this is really just constrained to those who can create branches here, e.g. non-critical infra and infra.